### PR TITLE
Fix report builder filter slugs

### PR DESCRIPTION
**Pull request by NoahCarnahan**
Fixes a bug that causes UCR query execution errors. http://manage.dimagi.com/default.asp?183456#1022893

The filter slug is eventually used as the name of a sqlalchemy bind param [in sqlagg](https://github.com/dimagi/sql-agg/blob/bb2f076e060ce438170831bdf0a945582324eca6/sqlagg/filters.py#L30). But, when this string is eventually converted into an action query for postgres, the bind parameter isn't properly delineated. e.g.
If your slug is `name-1`, the query ends up looking something like

```
SELECT ... FROM ... WHERE "name_column" = %(name)s-1 GROUP BY ...
```

The best place to solve this is probably in sqlagg (it should likely be using functions like [`bindparam()`](http://docs.sqlalchemy.org/en/rel_0_9/core/sqlelement.html#sqlalchemy.sql.expression.bindparam) instead of building the strings manually), but I didn't attempt to implement that given time constraints.

cc @kaapstorm 
FYI @snopoke who I think knows the most about sqlagg

### CODE REVIEW

File: corehq/apps/userreports/reports/specs.py [hunk 1]
```diff
@@ -297,13 +297,23 @@ def get_display(self):
...
+
+
 class FilterSpec(JsonObject):
     """
     This is the spec for a report filter - a thing that should show up as a UI filter element
     in a report (like a date picker or a select list).
     """
     type = StringProperty(required=True, choices=['date', 'numeric', 'choice_list', 'dynamic_choice_list'])
-    slug = StringProperty(required=True)  # this shows up as the ID in the filter HTML
+    slug = StringProperty(required=True, validators=_validate_filter_slug)  # this shows up as the ID in the filter HTML.
```
Comment (dimagimon) [inline corehq/apps/userreports/reports/specs.py]: E501 line too long (121 > 115 characters)

**Comment by snopoke** [comment]
Looks good, any possibility of tests?

File: corehq/apps/userreports/reports/specs.py [hunk 2]
```diff
@@ -297,13 +297,23 @@ def get_display(self):
         return self.display or self.value
 
 
+def _validate_filter_slug(s):
+    if "-" in s:
+        raise Exception(_(
+            """
+            Filter slugs must be legal sqlalchemy bind parameter names.
+            '-' and other special character are prohibited
```
Comment (NoahCarnahan) [inline corehq/apps/userreports/reports/specs.py]: Spent a while in the sqlalchemy docs but couldn't find any explanation of what characters are allowed or disallowed

**Comment by NoahCarnahan** [comment]
Added simple test

**Pull request merged**